extensions/markdown-language-features/src/test/markdownLink.test.ts

```diff
@@ -5,128 +5,245 @@
 import * as vscode from 'vscode';
 import * as assert from 'assert';
 import 'mocha';
-import { checkSmartPaste, createLinkSnippet } from '../languageFeatures/copyFiles/shared';
+import { SkinnyTextDocument, checkSmartPaste, createEditAddingLinksForUriList, createLinkSnippet } from '../languageFeatures/copyFiles/shared';
 import { validateLink } from '../languageFeatures/copyFiles/copyPasteLinks';
-
 suite('createEditAddingLinksForUriList', () => {
 
-	// end to end test of checkSmartPaste & createLinkSnippet
-	// check multicursor (end to end)
+	test('link to link snippet', async () => {
+		// createEditAddingLinksForUriList -> checkSmartPaste -> tryGetUriListSnippet -> createUriListSnippet -> createLinkSnippet
+
+		const skinnyDocument: SkinnyTextDocument = {
```

Comment:
As you add more tests, you may be able to use something like https://github.com/microsoft/vscode-markdown-languageservice/blob/1d9f6887521b3e3ce906595ef39a3e35d6d0e85c/src/test/inMemoryDocument.ts#L11 for this. Notice how it uses the `TextDocument` type from `vscode-languageserver-textdocument` which implements much of the API for you